src/visualization/streamlit-exp-v2.py

```diff
@@ -0,0 +1,108 @@
+'''
+streamlit-exp-v2.py
+'''
+import numpy as np
+import pandas as pd
+import streamlit as st
+from pathlib import Path
+import seaborn as sns
+from bokeh.plotting import figure
+from bokeh.models import ColumnDataSource, CategoricalColorMapper
+
+
+def gen_random_data():
+    '''
+    Generates random data to represent a high-dimensional data set.
+    dimensions: (1000, 1001)
+    '''
+    np.random.seed(8)
+    # generate column names, data and label names
+    column_names = ['col_' + str(x) for x in range(1, 769)]
+    data = np.random.uniform(-10, 10, size=(1000, 768))
+    label_names = ['label_' + str(x) for x in range(1, 21)]
+
+    # insert numpy array, return pandas dataframe
+    df = pd.DataFrame(data, columns=column_names)
+
+    # generate list of labels and randomly insert into pandas dataframe
+    labels = np.random.choice(label_names, size=len(df))
+    df.insert(loc=0, column='label', value=labels)
+
+    return df
+
+
+@st.cache
+def gather_umap_data():
+    cwd = Path.cwd()
+    filename1 = 'all_umap_pseudo_embeddings.json'
+    filename2 = 'centers_umap_pseudo_embeddings.json'
+    path_data = cwd/'data'/'current_centroid_data'
```

Comment:
You sure you want to hard code these path names in like this here? If you end up changing them at all this function will break. Consider moving these names to the `config.py` and importing them in. This goes for *any* path or filename that shows up in your source code. Hardcoding these is dangerous, because you might delete them, or rename them, or move them, in which case things will break.

Reply:
In my next branch, one of my first tasks will be to create a `config.py` file to avoid hardcoding these (amongst other things).